airmozilla/popcorn/views.py

```diff
@@ -106,6 +113,60 @@ def popcorn_data(request):
         }
 
 
+# Note that this view is publically available.
+# That means we can't trust the content but we can take it as a hint.
+@csrf_exempt
+@require_POST
+def vidly_webhook(request):
+    if not request.POST.get('xml'):
+        return http.HttpResponseBadRequest("no 'xml'")
+
+    xml_string = request.POST['xml'].strip()
+    try:
+        struct = xmltodict.parse(xml_string)
+    except ExpatError:
+        return http.HttpResponseBadRequest("Bad 'xml'")
+
+    try:
+        task = struct['Response']['Result']['Task']
+        migrate_submission(task)
```

Comment:
I still don't like the naming here. If it was called `migrate_user` you'd naturally thing the first argument into that function is a `user` thing. Not a `task`. I recommend you make the `migrate_submission` function oblivious of tasks. E.g. 
```python
migrate_submission(get_object_or_404(
    VidlySubmission,
    url=task['SourceFile'],
    tag=task['MediaShortLink']
))
```
And then change `migrate_submission` to something like this:
```python
def migrate_submission(submission):
    results = vidly.query(submission.tag)
    ...
```